src/Microsoft.AspNet.WebApi.Versioning/Dispatcher/DirectRouteControllerSelector.cs

```diff
@@ -52,6 +52,23 @@ internal override ControllerSelectionResult SelectController( ApiVersionControll
 
             var versionedController = GetVersionedController( aggregator, requestedVersion );
 
+           var versionedController = GetVersionedController( aggregator, requestedVersion );
+
+            if ( versionedController == null )
+            {
+                if(options.LookForLowerVesions && requestedVersion.MajorVersion.HasValue)
+                {
+                    for ( int i = requestedVersion.MajorVersion.Value - 1; i > 0; i-- )
+                    {
+                        versionedController = GetVersionedController( aggregator, new ApiVersion( i, 0 ) );
```

Comment:
Despite opting into this behavior, this is a gross over-simplification of what would be required to handle this feature in a reasonable way. The _minor_ version hasn't been considered. What happens if there is a _group version_ (aka date) or _status_ attached? The are simply not included. Furthermore, this only considers the _direct_ route scenario (aka attribute routing). It's quite possible that the one you're looking for is in the convention-based path. Mixing routing styles is a real (unfortunately) and supported scenario.